pylint/checkers/typecheck.py

```diff
@@ -579,6 +581,40 @@ def _emit_no_member(
     return True
 
 
+def _enum_has_attribute(owner, node) -> bool:
```

Comment:
```suggestion
def _enum_has_attribute(owner: Union[astroid.Instance, nodes.ClassDef], node: nodes.Attribute) -> bool:
```
As far as I know this seems to be the typing here.